src/utils/helpers.ts

```diff
@@ -154,49 +142,38 @@ export const getLatestDiscussionCommentId = (
     .reduce((a, b) => (a.node.createdAt > b.node.createdAt ? a : b))?.node
     .databaseId;
 
-export const getCommentId = (url: string) =>
-  /comments\/(?<id>\d+)/g.exec(url)?.groups?.id;
-
-export async function openInBrowser(
+export async function generateGitHubWebUrl(
   notification: Notification,
   accounts: AuthState,
 ) {
-  if (notification.subject.type === 'Release') {
-    getReleaseTagWebUrl(notification, accounts.token).then(({ url }) =>
-      openExternalLink(
-        generateGitHubWebUrl(
-          url,
-          notification.id,
-          accounts.user?.id,
-          undefined,
-        ),
-      ),
-    );
-  } else if (notification.subject.type === 'Discussion') {
-    getDiscussionUrl(notification, accounts.token).then(
-      ({ url, latestCommentId }) =>
-        openExternalLink(
-          generateGitHubWebUrl(
-            url || `${notification.repository.url}/discussions`,
-            notification.id,
-            accounts.user?.id,
-            latestCommentId
-              ? '#discussioncomment-' + latestCommentId
-              : undefined,
-          ),
-        ),
-    );
-  } else if (notification.subject.url) {
-    const latestCommentId = getCommentId(
+  let url = notification.repository.html_url;
+
+  if (notification.subject.latest_comment_url) {
+    url = await getHtmlUrl(
       notification.subject.latest_comment_url,
+      accounts.token,
     );
-    openExternalLink(
-      generateGitHubWebUrl(
-        notification.subject.url,
-        notification.id,
-        accounts.user?.id,
-        latestCommentId ? '#issuecomment-' + latestCommentId : undefined,
-      ),
-    );
+  } else if (notification.subject.url) {
+    url = await getHtmlUrl(notification.subject.url, accounts.token);
   }
+
+  // Perform any specific notification type handling (only required for a few special notification scenarios)
+  switch (notification.subject.type) {
```

Comment:
For defensive programming, should have a default case that throws a clear error that the type isn't handled.

Reply:
Is the new update better?  there's like a dozen ways these guards can be structured :p

Reply:
It is! Thanks for the cleanup :)